File: CS/Solution28.Module.Web/Controllers/ViewController1.cs
using System;
using System.ComponentModel;
using System.Collections.Generic;
using System.Diagnostics;
using System.Text;

using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using DevExpress.Persistent.Base;
using DevExpress.Data.Filtering;

namespace Solution28.Module.Web {
    public class ViewController1 : ViewController {
        public ViewController1() {
            ParametrizedRangeAction filterAction = new ParametrizedRangeAction(this, "FilterRange", "Unspecified", typeof(DateTime));
            filterAction.Execute += new ParametrizedActionExecuteEventHandler(filterAction_Execute);
        }
        private void filterAction_Execute(object sender, ParametrizedActionExecuteEventArgs e) {
            Range<DateTime> range = (Range<DateTime>)e.ParameterCurrentValue;
            ((ListView)View).CollectionSource.Criteria["By Date"] = CriteriaOperator.Parse("[Date] <= ? AND [Date] >= ?", range.To, range.From);
        }
    }
}

File: VB/Solution28.Module.Web/Controllers/ViewController1.vb
Imports System.ComponentModel
Imports System.Diagnostics
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.Data.Filtering

Namespace Solution28.Module.Web

    Public Class ViewController1
        Inherits ViewController

        Public Sub New()
            Dim filterAction As ParametrizedRangeAction = New ParametrizedRangeAction(Me, "FilterRange", "Unspecified", GetType(Date))
            AddHandler filterAction.Execute, New ParametrizedActionExecuteEventHandler(AddressOf filterAction_Execute)
        End Sub

        Private Sub filterAction_Execute(ByVal sender As Object, ByVal e As ParametrizedActionExecuteEventArgs)
            Dim range As Web.Range(Of Date) = CType(e.ParameterCurrentValue, Web.Range(Of Date))
            CType(View, ListView).CollectionSource.Criteria("By Date") = CriteriaOperator.Parse("[Date] <= ? AND [Date] >= ?", range.To, range.From)
        End Sub
    End Class
End Namespace
